data/json/materials.json

```diff
@@ -2624,14 +2624,22 @@
     "id": "charcoal",
     "name": "Charcoal",
     "copy-from": "wood",
-    "fuel_data": { "energy": "1600 kJ" }
+    "fuel_data": { "energy": "6240 kJ" },
+    "//": "physical density for charcoal estimated to sit at about 0.208 kg/liter, while energy density per kg is slightly lower than pure anthracite coal at 30 MJ/kg.",
+    "//1": "208g of charcoal (1 liter) yields 6240 kJ (30MJ x .208), and a single chunk (8mL/1.664g) is worth 50kJ",
+    "//2": "density sourced from https://www.aqua-calc.com/page/density-table/substance/charcoal and http://ukrfuel.com/news-physical-properties-of-charcoal-22.html",
+    "density": 0.208
   },
   {
     "type": "material",
     "id": "coal",
     "name": "Coal",
     "copy-from": "wood",
-    "fuel_data": { "energy": "30000 kJ" }
+    "//": "mostly pure anthracite coal, 100% purity anthracite has 33.5MJ/kilogram.",
+    "//1": "Raw density of anthracite is ~1.5g/cm^3 but bulk density is lower at slightly less than 1.0.",
+    "//2": "Roughly 5x the energy content of charcoal per liter.",
+    "fuel_data": { "energy": "31200 kJ" },
+    "density": 0.931
```

Comment:
Use the base density value (1.5g/cm³) here, but calculate the mass/volume of the item itself based on bulk density.

I kind of [touched on it before](https://github.com/CleverRaven/Cataclysm-DDA/pull/62180#discussion_r1096909315):

> [use bulk density] when we are trying to determine the weight/volume properties that one unit of an item should have, since this is what determines how much space it will take up in inventory/container
> [use true density] in materials.json for the purposes of density sanity checks -- it should never be exceeded.

For example, material for a metal such as steel is defined with true density of a solid block, and the true density is used for an item that is a solid ingot of the metal. Whereas the item for small pieces of crumpled scrap metal has mass & volume stats that [reflect bulk density](https://github.com/CleverRaven/Cataclysm-DDA/blob/9de4b9c1828254be0e4148adc576a0f9f327caf0/data/json/items/resources/metal.json#L34) for that specific form factor.

If you use bulk density here, it could preclude potential future items. For example, brick-like charcoal blocks are a thing that exists IRL. Since it is a contiguous block (not exactly *solid*, since material is somewhat porous) and multiples of it would stack well together, if we create that item in the game it would have mass/vol close to true density of charcoal material. But if you put bulk density in materials.json, then the "charcoal brick" item would trip the bad density sanity test, which would consider it to be too heavy.

Re: a reasonable value for charcoal, the ukrfuel source you mentioned before gives a range of
> 0.2 and 0.6 t/m3

so for the purposes of materials.json it's safest to take the upper limit and use (0.6g/cm³)

Reply:
OK, I left additional comments in materials.json. For now I'd rather leave its energy data as assuming non-bulk density because it's important that a fuel bunker full of coal represents that physical density. We would probably need a second material in game anyways for compressed, denser charcoal because fuel bunkers care about the volume, rather than the weight, of their contents. 

If we were to adjust the present fuel density of charcoal it would be pretty easy to reduce it to 4 ml per or whatever, all that would be needed to do would be making the cardboard boxes it spawns in come with more charcoal than before.